### ISSUE TITLE
electron app does not work on macOS 13.4

### ISSUE THREAD

**Issue by hiroMTB** [thomasgeissl/popeye#21]
Hello Thomas, thanks for your great work.
Just reporting that version 0.0.3 doesn't work on macOS 13.4. 

![Screenshot 2023-10-12 at 18 10 05](https://github.com/thomasgeissl/popeye/assets/193280/7a05bf4c-ec46-4573-bbf5-000e3497444e)

but if I compile code, it works well.

**Comment by thomasgeissl** [comment]
hey thanks for message and sorry, i havent seen it earlier. 
did you figure out how to fix it? i am quite busy with work, but can have a look once a have some free time.

**Comment by hiroMTB** [comment]
I end up making own MediaPipe OSC Sender because I couldn't fix the issue, and also MediaPipe is updated. I'm also bit busy at the moment but would be nice to have time to contribute updating Popeye later. Thanks for your message!

**Comment by thomasgeissl** [comment]
i will hopefully find some time over christmas, planning to updat ethe lib. there seems to be an issue with the electron bundle. it is way to big.
potentially will port it to tauri.

**Comment by hiroMTB** [comment]
Cool, never heard of tauri but looks interesting.

**Comment by thomasgeissl** [comment]
i started porting it to tauri, which now reduces the bundle size to 20mb instead of 600mb.
still some fixes left to do. 

in case you wanna give it try. let me know how it works for you.
i also will cleanup the js part soon, it got a bit messy. and should update the media pipe libs.
any contributions are welcome of course.

**Comment by hiroMTB** [comment]
Cool Thomas! I will try it out!

Last year, we made our own webApp that communicates with oF via websocket
(without UDP). I will ask if we can publish the app.

Recent updates of MediaPipe improves reducing jittering and adds new
parameters (visibility, if I remember correctly).

**Comment by thomasgeissl** [comment]
sounds good. 
and quite related, i started working on an webview for oF. that helps to get rid of the ws communication.
basically an oF based electron.
https://github.com/thomasgeissl/ofxChoc/tree/main/example-webview

**Comment by hiroMTB** [comment]
I just tried ofxChoc and all works great on macOS 14.
So I tried replacing /dist contents of example-webserver_webview example with our MediaPipe webApp.
and WOW! Magic!

**Comment by thomasgeissl** [comment]
thanks for giving it a try and reporting back. glad to hear that it worked without issues.
the event system might need some internal tweakings, but i think the basics are already there.

**Comment by hiroMTB** [comment]
Webcam works, MediaPipe works, Landmark detection works, Gui works!
One issue I found is that video doesn't record. (We have a HTML5 video recording feature for webcam input to adjust MediaPipe parameters). Great work!

**Comment by thomasgeissl** [comment]
thanks. feel free to add an issue on the other repo and move that conversation over there. 
probably nothing i can fix easily, maybe more an webkit issue or how it is implemented in choc. wonder if it works on windows.

**Comment by hiroMTB** [comment]
btw, I'm quite surprised to see that oF's Project Generator is about 430MB. 
It might be good to re-write with Tauri.

**Comment by thomasgeissl** [comment]
yeah, electron is quite heavy. also it takes a bit longer to load the app.
i would like to see it written with ofxChoc, since it has a c++ backend instead of rust.

**Comment by hiroMTB** [comment]
oh i see, that's also sounds good.

**Comment by thomasgeissl** [comment]
electron -> tauri -> oF

since i am more comfortable with c++ than rust.
here is the version with an openFrameworks backend.
https://github.com/thomasgeissl/ofxChoc/tree/main/example-popeye

i should remove the empty oF window and bundle the frontend with app, or at least auto download it from github (releases, or gh pages version). also, i wonder if there is an easy way to download, cache and serve the frontend part from the c++ side. maybe similar to how PWAs do that. could be nice to have at least the frontend auto update.
